packages/@react-aria/collections/src/CollectionBuilder.tsx

```diff
@@ -109,7 +115,7 @@ function useCollectionDocument<T extends object, C extends BaseCollection<T>>(cr
     return collection;
   }, [document]);
   let getServerSnapshot = useCallback(() => {
-    document.isSSR = true;
+    document.setSSR(true);
```

Comment:
had to introduce this because the React compiler is a bit strict about mutating, I think it should be allowed in this case, but React doesn't know that, so I've tricked it. I had tried storing the document in a ref and editing it that way, which fixes the compiler problem, but it broke some tests, I'm not sure why and it makes me a little worried. This is the closest to what we've tested for years though, so I'm ok with this for now.